src/main/java/de/flapdoodle/embed/process/store/CachedPostgresArtifactStore.java

```diff
@@ -44,8 +45,12 @@ public IExtractedFileSet extractFileSet(Distribution distribution) throws IOExce
         try {
             final File dir = this.eDir.asFile();
             final FileSet filesSet = downloadConfig.getPackageResolver().getFileSet(distribution);
-            if (dir.exists() && dir.isDirectory() && filesSet.entries().stream()
-                    .allMatch(entry -> exists(get(dir.getPath(), "pgsql", "bin")))) {
+            final Path path = get(dir.getPath(),
+                                  "pgsql" + "-" + distribution.getVersion().asInDownloadPath(), "bin");
+            if (dir.exists()
+                && dir.isDirectory()
+                && filesSet.entries().stream().allMatch(entry -> exists(path))
```

Comment:
Now I can see that this condition was created incorrectly. In fact it is redundant and can be safely replaced with `exists(path)`.